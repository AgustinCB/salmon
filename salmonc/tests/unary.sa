!true;
!false;
-3;
+-3;
