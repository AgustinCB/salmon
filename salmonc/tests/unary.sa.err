Constants: [Bool(true), Bool(false), Integer(0), Integer(3), String(0)]
Instructions: [Instruction { instruction_type: Constant(0), location: 0 }, Instruction { instruction_type: Not, location: 0 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Not, location: 1 }, Instruction { instruction_type: Constant(2), location: 2 }, Instruction { instruction_type: Constant(3), location: 2 }, Instruction { instruction_type: Minus, location: 2 }, Instruction { instruction_type: Constant(2), location: 3 }, Instruction { instruction_type: Constant(3), location: 3 }, Instruction { instruction_type: Minus, location: 3 }, Instruction { instruction_type: Abs, location: 3 }]
Locations: [Location { address: 0, line: 0 }, Location { address: 0, line: 1 }, Location { address: 0, line: 2 }, Location { address: 0, line: 3 }]
