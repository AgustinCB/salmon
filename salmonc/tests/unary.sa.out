0 - Integer(3)
1 - Integer(-3)
2 - Bool(true)
3 - Bool(false)
